src/lib/getTasks.ts

```diff
@@ -7,30 +7,104 @@ import {
     IRead,
 } from "@rocket.chat/apps-engine/definition/accessors";
 import { IRoom } from "@rocket.chat/apps-engine/definition/rooms";
-import { IUIKitResponse, TextObjectType, UIKitBlockInteractionContext } from '@rocket.chat/apps-engine/definition/uikit';
-
+import { IUIKitResponse, TextObjectType, UIKitViewSubmitInteractionContext , ButtonStyle} from '@rocket.chat/apps-engine/definition/uikit';
 import {
     ISlashCommand,
     SlashCommandContext,
 } from "@rocket.chat/apps-engine/definition/slashcommands";
+import { IUIKitViewSubmitIncomingInteraction } from "@rocket.chat/apps-engine/definition/uikit/UIKitIncomingInteractionTypes";
+import { ICreateTaskState } from "../facade/IClickUpService";
+import { IUser } from "@rocket.chat/apps-engine/definition/users";
+import { getAccessTokenForUser } from "../storage/users";
+import { ModalsEnum } from "../enums/Modals";
+import { MiscEnum } from "../enums/Misc";
 
 export async function getTasks({
     context,
+    data,
     room,
     read,
     persistence,
     modify,
     http,
 }: {
-    context: UIKitBlockInteractionContext,
+    context: UIKitViewSubmitInteractionContext,
+    data: IUIKitViewSubmitIncomingInteraction,
     room: IRoom;
     read: IRead;
     persistence: IPersistence;
     modify: IModify;
     http: IHttp;
 }) {
-    
-    //to be implemented next week.
+    const state = data.view.state;
+    const user: IUser = context.getInteractionData().user;
+    const token = await getAccessTokenForUser(read, user);
+    const list_id = state?.[ModalsEnum.LIST_ID_BLOCK]?.[ModalsEnum.LIST_ID_INPUT];
+    const archived = state?.[ModalsEnum.ARCHIVED_BLOCK]?.[ModalsEnum.ARCHIVED_ACTION_ID] == "Yes"?"true":"false";
+    const subtasks = state?.[ModalsEnum.SUBTASKS_BLOCK]?.[ModalsEnum.SUBTASKS_ACTION_ID] == "Yes"?"true":"false";
+    const taskLimit = state?.[ModalsEnum.TASK_LIMIT_BLOCK]?.[ModalsEnum.TASK_LIMIT_INPUT];
+    var limit = taskLimit;
```

Comment:
please, avoid using var as much as possible, switch to let that it will be better.

for more reference:
https://medium.com/@codingsam/awesome-javascript-no-more-var-working-title-999428999994